client/src/actions/users.js

```diff
@@ -7,10 +7,10 @@ import { ADD_USER_DATA } from "./types";
 /**
  * This action creator will be called when a user registers.
  *
- * @param userData object containing user email, given and family name and password
+ * @param userData object containing user email, team and organization name and password
  * @returns a thunk responsible for posting the data to the api and dispatching
  */
-export const addUserAction = (userData) => async (dispatch) => {
+export const addUserAction = (userData, apiResult) => async (dispatch) => {
```

Comment:
Rather than taking an `apiResult` callback for signalling errors, I think it's better to dispatch separate actions (with different types) accordingly.
@pmyip3 did something similar: https://github.com/Researchify/Researchify/blob/scholar/client/src/actions/types.js#L22-L25